HW_9/task 9.2.js

```diff
@@ -0,0 +1,158 @@
+// 6. Добавьте метод addWorker(worker), добавляющий работника в массив workers. 
+//     На вход метод должен принимать объект класса Worker. 
+//     Если объект не является инстансом класса Worker - выкинуть ошибку
+// 7. Добавьте метод addAnimal(animal), добавляющий животное в массив animals.
+//     На вход метод должен принимать объект класса Animal, как и любого из его наследников. 
+//     Если объект не является инстансом класса Animal - выкинуть ошибку
+//     ТАКЖЕ, если объект является инстансом класса Snake - выкинуть ошибку с тексом "There will be no snakes, mister Potter!"
+// 8. Добавьте методы removeWorker() и removeAnimal() // Подумайте, как будем удалять, по какому полю будем выбирать:)
+
+class Animal {
+    constructor (type, color, weight, height, place_of_origin) {
+      this.type = type;
+      this.color = color;
+      this.weight = weight;
+      this.height = height;
+      this.place_of_origin = place_of_origin;
+    }
+    getInfo() {
+      return `Type of animal - ${this.type}, color - ${this.color}, weight - ${this.weight} kg, height - ${this.height} cm, place of origin - ${this.place_of_origin}`;
+    }
+    get color() {
+      return this._color;
+    }
+    set color(newColor) {
+      const validColors = ['red', 'black', 'white', 'blue'];
+      newColor.toLowerCase();
+      if (!validColors.includes(newColor)) throw new Error('Wrong color');
+      this._color = newColor;
+    }
+}
+  
+class Snake extends Animal {
+    #isPoisonous;
+    constructor (type, color, weight, height, place_of_origin, isPoisonous) {
+      super(type, color, weight, height, place_of_origin);
+      this.#isPoisonous = isPoisonous;
+    }
+    checkPoisonous() {
+      if (this.#isPoisonous) return `${this.type} is poisonous`;
+      return `${this.type} is not poisonous`;
+    }
+}
+
+class Bird extends Animal {
+    #isFlying;
+    constructor(type, color, weight, height, place_of_origin, isFlying) {
+        super(type, color, weight, height, place_of_origin);
+        this.#isFlying = isFlying;
+    }
+    checkFlying() {
+        if (this.#isFlying) return `${this.type} is flying`;
+        return `${this.type} is not flying`;
+    }
+}
+
+class CatLike extends Animal {
+    constructor(type, color, weight, height, place_of_origin, isSafeToPet) {
+        super(type, color, weight, height, place_of_origin);
+        this.isSafeToPet = isSafeToPet;
+    }
+    checkSafeToPet() {
+        if (this.isSafeToPet) return `${this.type} is safe to pet`;
+        return `${this.type} is not safe to pet`;
+    }
+}
+
+class Worker {
+    constructor(firstName, lastName, phone) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.phone = phone;
+    }
+    getFullName() {
+        return `${this.firstName} ${this.lastName}`;
+    }
+}
+
+class Zoo {
+    constructor(address, title, ticket_price) {
+        this.address = address;
+        this.title = title;
+        this.ticket_price = ticket_price;
```

Comment:
In constructor you've make properties _public_ `this.address`
In get/set accessor you're using the underscore `this._address` what tells us that these properties are _protected_
Think it's better to use one thing